### PR TITLE
Fix 23198: rework event intro

### LINKED ISSUE #23198: Duplicate content inside the JavaScript events page
### MDN URL

https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events

### What specific section or headline is this issue about?

https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#a_simple_example

### What information was incorrect, unhelpful, or incomplete?

The explanation of `addEventListener()` method is being repeated on different sections on the same page.

There is a section titled "A simple example" that has the following content:

> ```
> const btn = document.querySelector('button');
> 
> function random(number) {
>   return Math.floor(Math.random() * (number+1));
> }
> 
> btn.addEventListener('click', () => {
>   const rndCol = `rgb(${random(255)}, ${random(255)}, ${random(255)})`;
>   document.body.style.backgroundColor = rndCol;
> });
> ```
> Objects that can fire events have an [addEventListener()](https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener) method, that takes at least two arguments: the name of the event and a function to handle the event. So we call the button's addEventListener() method, passing in:
> 
> - the string 'click', to indicate that we want to listen to the click event
> 
> - a function to call when the event happens. In our case, the function generates a random RGB color and sets the page [<body>](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/body) [background-color](https://developer.mozilla.org/en-US/docs/Web/CSS/background-color) equal to that color.

Another section titled "Using addEventListener()" has almost the same content:

> ```
> const btn = document.querySelector('button');
> 
> function random(number) {
>   return Math.floor(Math.random() * (number+1));
> }
> 
> btn.addEventListener('click', () => {
>   const rndCol = `rgb(${random(255)}, ${random(255)}, ${random(255)})`;
>   document.body.style.backgroundColor = rndCol;
> });
> ```
> 
> Inside the addEventListener() function, we specify two parameters: the name of the event we want to register this handler for, and the code that comprises the handler function we want to run in response to it.

### What did you expect to see?

The "Using addEventListener()" section begins with an explanation of the `addEventListener()` method and an explanation on how to use a separate named function for handling the event instead of using an anonymous function.

I would suggest removing the duplicate content inside the "Using addEventListener()" section and replacing it with something like this:

> We've already seen how to handle a click event by using an anonymous function from the previous example. But 
> we could also use a separate named function to handle the same event, like so:
> 
> ```
> const btn = document.querySelector('button');
> 
> function random(number) {
>   return Math.floor(Math.random() * (number+1));
> }
> 
> function changeBackground() {
>   const rndCol = `rgb(${random(255)}, ${random(255)}, ${random(255)})`;
>   document.body.style.backgroundColor = rndCol;
> }
> 
> btn.addEventListener('click', changeBackground);
> ```

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### PR DESCRIPTION
**Pull request by wbamberg**
https://github.com/mdn/content/issues/23198 reports that the first part of https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events duplicates the sample code and explanation.

I think the idea here is that the first bit introduces the concepts with an example, while the second bit introduces the API itself. So I've changed things a bit so we only talk about the code in the second bit, and added pointers to acknowledge that it is the same code in both places.

I made a few other changes:
- I think "A series of fortunate events" is an obscure reference for many people, so removed it
- I moved "It's not just web pages" so the example flows directly into the code explanation, and simplified "It's not just web pages" so we can get going faster.

Prettier changes are in https://github.com/mdn/content/pull/23224/commits/f0b942a39a8a927150a36f73497460fbbe2709c5, substantive changes are in https://github.com/mdn/content/pull/23224/commits/f89210b7fabb4f8869e31b52a6179dd0c02931a6.

### CODE REVIEW

File: files/en-us/learn/javascript/building_blocks/events/index.md [hunk 1]
```diff
@@ -68,10 +68,18 @@ The listener listens out for the event happening, and the handler is the code th
...
 
-### A simple example
+### It's not just web pages
+
+Events are not unique to JavaScript — most programming languages have some kind of event model, and the way the model works often differs from JavaScript's way.
+
+In fact, the event model in JavaScript for web pages differs from the event model for JavaScript as it is used in other environments. For example, [Node.js](/en-US/docs/Learn/Server-side/Express_Nodejs) is a very popular JavaScript runtime that enables developers to use JavaScript to build network and server-side applications.
+The [Node.js event model](https://nodejs.org/docs/latest-v12.x/api/events.html) is quite different from the event model in web pages.
+
+You don't need to understand anything about other such environments at this stage; we just wanted to make it clear that events can differ in different programming environments.
```
Comment (Josh-Cena) [inline files/en-us/learn/javascript/building_blocks/events/index.md:78]: Is the information before mentioned on any other page, though? I'm a bit sorry about its loss. Maybe move this section towards the end of the page as an afterthought, so we have more space for random facts?

**Comment by wbamberg** [comment]
@Josh-Cena , could you take another look at this one please?

File: files/en-us/learn/javascript/building_blocks/events/index.md [hunk 2]
```diff
@@ -774,6 +767,21 @@ The output is as follows (try clicking around on it):
...
 
+## It's not just web pages
+
+Events are not unique to JavaScript — most programming languages have some kind of event model, and the way the model works often differs from JavaScript's way.
+In fact, the event model in JavaScript for web pages differs from the event model for JavaScript as it is used in other environments.
+
+For example, [Node.js](/en-US/docs/Learn/Server-side/Express_Nodejs) is a very popular JavaScript runtime that enables developers to use JavaScript to build network and server-side applications.
+The [Node.js event model](https://nodejs.org/docs/latest-v12.x/api/events.html) relies on listeners to listen for events and emitters to emit events periodically — it doesn't sound that different, but the code is quite different, making use of functions like `on()` to register an event listener, and `once()` to register an event listener that unregisters after it has run once.
+The [HTTP connect event docs](https://nodejs.org/docs/latest-v12.x/api/http.html#http_event_connect) provide a good example.
```
Comment (Josh-Cena) [inline files/en-us/learn/javascript/building_blocks/events/index.md:777]:
```suggestion
The [Node.js event model](https://nodejs.org/api/events.html) relies on listeners to listen for events and emitters to emit events periodically — it doesn't sound that different, but the code is quite different, making use of functions like `on()` to register an event listener, and `once()` to register an event listener that unregisters after it has run once.
The [HTTP connect event docs](https://nodejs.org/api/http.html#event-connect) provide a good example.
```

File: files/en-us/learn/javascript/building_blocks/events/index.md [hunk 3]
```diff
@@ -83,85 +82,64 @@ button {
...
   document.body.style.backgroundColor = rndCol;
 });
 ```
 
-> **Note:** You can find the [full source code](https://github.com/mdn/learning-area/blob/main/javascript/building-blocks/events/random-color-addeventlistener.html) for this example on GitHub (also [see it running live](https://mdn.github.io/learning-area/javascript/building-blocks/events/random-color-addeventlistener.html)).
+The HTML {{HTMLElement("button")}} element will fire an event when the user clicks the button. So it defines an `addEventListener()` function, which we are calling here. We're passing in two parameters:
 
-Inside the `addEventListener()` function, we specify two parameters: the name of the event we want to register this handler for, and the code that comprises the handler function we want to run in response to it.
+- the string `"click"`, to indicate that we want to listen to the click event. Buttons can fire lots of other events, such as [`"mouseover"`](/en-US/docs/Web/API/Element/mouseover_event) when the user moves their mouse over the button, or [`"keydown"`](/en-US/docs/Web/API/Element/keydown_event) when use user presses a key and the button is focused.
```
Comment (Josh-Cena) [inline files/en-us/learn/javascript/building_blocks/events/index.md:125]:
```suggestion
- the string `"click"`, to indicate that we want to listen to the click event. Buttons can fire lots of other events, such as [`"mouseover"`](/en-US/docs/Web/API/Element/mouseover_event) when the user moves their mouse over the button, or [`"keydown"`](/en-US/docs/Web/API/Element/keydown_event) when the user presses a key and the button is focused.
```

File: files/en-us/learn/javascript/building_blocks/events/index.md [hunk 4]
```diff
@@ -83,85 +82,64 @@ button {
...
-The event model is similar to the web events model, but a bit different — event listeners' properties are camel-cased (such as `onMessage` rather than `onmessage`), and need to be combined with the `addListener` function.
-See the [`runtime.onMessage`](/en-US/docs/Mozilla/Add-ons/WebExtensions/API/runtime/onMessage#examples) page for an example.
-
-You don't need to understand anything about other such environments at this stage in your learning; we just wanted to make it clear that events can differ in different programming environments.
+{{ EmbedLiveSample('An example: handling a click event', '100%', 200, "", "") }}
 
 ## Using addEventListener()
 
-The recommended mechanism for adding event handlers in web pages is the [`addEventListener()`](/en-US/docs/Web/API/EventTarget/addEventListener) method:
+As we saw in the last example, objects that objects that can fire events have an {{domxref("EventTarget/addEventListener", "addEventListener()")}} method, and this is the recommended mechanism for adding event handlers.
```
Comment (Josh-Cena) [inline files/en-us/learn/javascript/building_blocks/events/index.md:106]:
```suggestion
As we saw in the last example, objects that can fire events have an {{domxref("EventTarget/addEventListener", "addEventListener()")}} method, and this is the recommended mechanism for adding event handlers.
```

**Review by Josh-Cena — APPROVED** [review]
Mostly LGTM, so approving

**Pull request merged**